packages/flutter/lib/src/rendering/unconstrained_box.dart

```diff
@@ -0,0 +1,115 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+
+import 'box.dart';
+import 'debug_overflow_indicator.dart';
+import 'object.dart';
+import 'shifted_box.dart';
+import 'stack.dart';
+
+/// Renders a box, imposing no constraints on its child, allowing the child to
+/// render at its "natural" size.
+///
+/// This allows a child to render at the size it would render if it were alone
+/// on an infinite canvas with no constraints. This container will then expand
+/// as much as it can within its own constraints and align the child based on
+/// [alignment].  If the container cannot expand enough to accommodate the
+/// entire child, the child will be clipped.
+///
+/// In debug mode, if the child overflows the container, a warning will be
+/// printed on the console, and black and yellow striped areas will appear where
+/// the overflow occurs.
+///
+/// See also:
+///
+///  * [ConstrainedBox] for a box which imposes constraints on its child.
+///  * [Container], a convenience widget that combines common painting,
+///    positioning, and sizing widgets.
+///  * [OverflowBox], a widget that imposes different constraints on its child
+///    than it gets from its parent, possibly allowing the child to overflow
+///    the parent.
+class RenderUnconstrainedBox extends RenderAligningShiftedBox with DebugOverflowIndicatorMixin {
+  RenderUnconstrainedBox({
+    RenderBox child,
+    @required TextDirection textDirection,
+    @required AlignmentGeometry alignment,
+  }) : assert(alignment != null),
+       super.mixin() {
+    this.textDirection = textDirection;
+    this.alignment = alignment;
+    this.child = child;
```

Comment:
it would be more efficient to use arguments to the super constructor here rather than go through the setters